### ISSUE TITLE
Confused about RGB and BGR transformation for to_rgb setting in config. 

### ISSUE THREAD

**Issue by chulminkw** [open-mmlab/mmdetection#5214]
Hello and Thanks for this great object detection packages. 

I have a question about RGB and BGR changes in to_rgb setting in config file. 

I checked into configs/faster_rcnn/faster_rcnn_r50_fpn_1x_coco.py and found to_rgb=True
which seems like pytorch style config file. 
Followings seems image normalization and transformation. 

img_norm_cfg = dict(
    mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=True)
train_pipeline = [
    .......
    dict(
        type='Normalize',
        mean=[123.675, 116.28, 103.53],
        std=[58.395, 57.12, 57.375],
        to_rgb=True),
]

test_pipeline = [
     dict(
        .....
        transforms=[
          ........
            dict(
                type='Normalize',
                mean=[123.675, 116.28, 103.53],
                std=[58.395, 57.12, 57.375],
                to_rgb=True),
  ])

How does to_rgb setting work in preprocessing transformation? 
Does it mean the input image array is BGR ordered and to_rgb setting is going to change BGR to RGB and apply mean/std transformation after that? or input image array is expected to have RGB ordered ? 

In addition to that, when I tested with inference-only model with faster_rcnn_r50_fpn_1x_coco.py with the following code

##### start of code #####
from mmdet.apis import init_detector, inference_detector
import mmcv
import cv2

config_file = 'configs/faster_rcnn/faster_rcnn_r50_fpn_1x_coco.py'
checkpoint_file = 'checkpoints/faster_rcnn_r50_fpn_1x_coco_20200130-047c8118.pth'
model = init_detector(config_file, checkpoint_file, device='cuda:0')

'''BGR ordered array '''
img_arr = cv2.imread('/content/mmdetection/demo/demo.jpg')
results = inference_detector(model, img_arr)

show_result_pyplot(model, img_arr, results) 
#### end of code

The input image array should be BGR ordered, then Output image is RGB ordered. If the input image array is RGB ordered then output image is BGR ordered. 

I tested with other configuration like faster_rcnn_r50_caffe_fpn_1x_coco.py which is to_rgb=False in train and test pipeline(which  different from previous case). But the result was same. the input image array should be BGR ordered and output image is RGB ordered. 

How can I interpret this case? Is it related with test_pipeline to_rgb setting?(or  just show_result_pyplot() issue)

Best regards

**Comment by ZwwWayne** [comment]
This is related to the pre-trained models. We may add documentation in the future to provide more details about that.

**Comment by chulminkw** [comment]
Thanks ZwwWayne,
